jplotter/src/main/java/hageldave/jplotter/renderables/Legend.java

```diff
@@ -742,6 +742,14 @@ public void renderSVG(Document doc, Element parent, int w, int h) {
 		if(!isEnabled()){
 			return;
 		}
+		if(w == 0 || h == 0){
+			return;
+		}
+		if(isDirty() || viewPortWidth != w || viewPortHeight != h){
+			viewPortWidth = w;
+			viewPortHeight = h;
+			updateGL(false); // only clearGL requires GL context, but all GL resources are null, so no prob.
```

Comment:
GL resources will only be null when using fallback rendering, otherwise we will run into problems.

Using clearGL() was neat/lazy because it just throws away all the points, lines, triangle and text objects, so that setup can create new ones. But when these objects are created with regular GL rendering, they need to be closed when a GL conetxt is active before throwing them away. Otherwise the GL resources keep existing in memory even after the garbage collector got rid of the enclosing object.

Okay, so what I would suggest here is renaming the method, maybe just call it `clear`, and instead of calling `close()` on the Closeable objects, it collects them in a list `List<Closeable> toCloseLater`. These objects then need to be closed during regular rendering in the `render` and `renderFallback` method, and also in `close()`.

Maybe I can implement these changes myself since you're not used to this GL stuff. Good to have my thoughts documented here anyways.